extension/healthcheckextension/healthcheckextension_test.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package healthcheckextension
+
+import (
+	"net"
+	"net/http"
+	"runtime"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-service/extension/extensiontest"
+	"github.com/open-telemetry/opentelemetry-service/internal/testutils"
+)
+
+func TestHealthCheckExtensionUsage(t *testing.T) {
+	config := Config{
+		Port: getAvailablePort(t),
+	}
+
+	hcExt, err := newServer(config, zap.NewNop())
+	require.NoError(t, err)
+	require.NotNil(t, hcExt)
+
+	mh := extensiontest.NewMockHost()
+	require.NoError(t, hcExt.Start(mh))
+	defer hcExt.Shutdown()
+
+	// Give a chance for the server goroutine to run.
+	runtime.Gosched()
+
+	client := &http.Client{}
+	url := "http://localhost:" + strconv.Itoa(int(config.Port))
+	resp0, err := client.Get(url)
+	require.NoError(t, err)
+	defer resp0.Body.Close()
+
+	require.Equal(t, http.StatusServiceUnavailable, resp0.StatusCode)
+
+	hcExt.Ready()
+	resp1, err := client.Get(url)
+	require.NoError(t, err)
+	defer resp1.Body.Close()
+	require.Equal(t, http.StatusNoContent, resp1.StatusCode)
```

Comment:
This is how this component behaves since we started using it on OC and it makes sense: you give the status but doesn't bother in sending the body, the message makes that clear.

Reply:
gah - an old version of the jaeger pkg hc returned a 204 https://github.com/jaegertracing/jaeger/blob/v1.9.0/pkg/healthcheck/handler.go#L78 well mystery and oddity solved.